lib/workers/repository/update/pr/code-owners.ts

```diff
@@ -21,37 +32,129 @@ export async function codeOwnersForPr(pr: Pr): Promise<string[]> {
 
     logger.debug(`Found CODEOWNERS file: ${codeOwnersFile}`);
 
+    // Get list of modified files in PR
     const prFiles = await getBranchFiles(pr.sourceBranch);
-    const rules = codeOwnersFile
+
+    if (!prFiles) {
+      logger.debug('PR includes no files');
+      return [];
+    }
+
+    // Convert CODEOWNERS file into list of matching rules
+    const fileOwnerRules = codeOwnersFile
       .split(newlineRegex)
+      // Remove empty and commented lines
       .map((line) => line.trim())
       .filter((line) => line && !line.startsWith('#'))
+
+      // Extract pattern & usernames
       .map((line) => {
         const [pattern, ...usernames] = line.split(regEx(/\s+/));
         return {
           usernames,
-          match: (path: string) => {
-            const matcher = ignore().add(pattern);
-            return matcher.ignores(path);
-          },
+          pattern,
+          match: (path: string) => ignore().add(pattern).ignores(path),
```

Comment:
how expensive is `ignore().add(pattern)`? we could create a variable before the return, so it will kept as reference and we skip the initializing on every match call.

Reply:
Solved in b8fa2da6f9879d2a4d7987ba810280634d1068aa